### DISCUSSION TITLE
categorizing messages?

### DISCUSSION THREAD

**Discussion by DamianPacanowski** [PHPMailer/PHPMailer#2990] [Ideas]
Hello and welcome. Have you thought about categorizing the messages you send? 
Just like gmail, for example, categorizes messages into main, notifications, offers, forums and etc?
`$mail->Category = 'you enter here for example "notifications" ';`
For example, when Gmail receives mail, it would immediately assign it to a given folder (Category)
Regards

**Comment by Synchro** [comment]
This isn't something that ca be controlled from the sending end. Most email clients have the ability to filter messages in various ways, but it's entirely up to the recipient how it works. For your example, you could achieve that with a custom header (the `X-` prefix is traditionally used to denote non-standard headers):

    $mail->addCustomHeader('X-Category', 'notifications');

and then filter on that header in your mail client.

**Reply by DamianPacanowski** [reply]
following the instructions, adding this
`$mail->addCustomHeader('X-category', 'updates');`
should filtering of messages through gmail mailboxes
but he doesn't do that
I tried this too
`$mail->addCustomHeader('category', 'updates');`
`$mail->addCustomHeader('X', 'category:updates');`
[https://support.google.com/mail/answer/7190?hl=en](https://support.google.com/mail/answer/7190?hl=en)

**Reply by Synchro** [reply]
Well, that's something I didn't know - gmail's filtering is pretty bad! Most email clients can filter on header contents, but as I said, this is down to what's implemented in the client, not the sender. To downgrade your ambitions a little, you could instead put a prefix on the subject line, and that will work better with gmail's filters because it has an option for matching words in the subject.
